pkg/gitops.go

```diff
@@ -343,22 +355,30 @@ func CloneRepo(outputPath string, remote string, componentName string, branch st
 // 2. componentName: The component name corresponding to a single Component in an Application. eg. component.Name
 // 3. The executor to use to execute the git commands (either gitops.executor or gitops.mockExecutor)
 // 4. The path within the repository to generate the resources in
-func RemoveComponent(outputPath string, componentName string, context string) error {
+func (s Gen) RemoveComponent(outputPath string, componentName string, context string) error {
 	repoPath := filepath.Join(outputPath, componentName)
 	gitopsFolder := filepath.Join(repoPath, context)
 	componentPath := filepath.Join(gitopsFolder, "components", componentName)
-	if out, err := execute(repoPath, rmCommand, "-rf", componentPath); err != nil {
+	if out, err := execute(repoPath, RmCommand, "-rf", componentPath); err != nil {
 		return fmt.Errorf("failed to delete %q folder in repository in %q %q: %s", componentPath, repoPath, string(out), err)
 	}
 	return nil
 }
 
 // GetCommitIDFromRepo returns the commit ID for the given repository
-func GetCommitIDFromRepo(fs afero.Afero, repoPath string) (string, error) {
+func (s Gen) GetCommitIDFromRepo(fs afero.Afero, repoPath string) (string, error) {
 	var out []byte
 	var err error
-	if out, err = execute(repoPath, gitCommand, "rev-parse", "HEAD"); err != nil {
+	if out, err = execute(repoPath, GitCommand, "rev-parse", "HEAD"); err != nil {
 		return "", fmt.Errorf("failed to retrieve commit id for repository in %q %q: %s", repoPath, string(out), err)
 	}
 	return string(out), nil
 }
+
+// NewGitopsGen returns a Generator implementation
+func NewGitopsGen() Gen {
+	return Gen{}
+}
+
+type Gen struct {
+}
```

Comment:
quick nit: Can we have this near the top of the file, I was wondering what `Gen` was as I going thru the file but fine otherwise